classes/local/skel/lib_php_file.php

```diff
@@ -66,4 +66,26 @@ public function add_supported_feature($feature) {
 
         $this->data['self']['supports'][] = $feature;
     }
+
+    /**
+     * Returns the variables needed for rendering the template.
+     *
+     * @param string $plugintype.
+     * @return string[].
+     */
+    static public function get_template_variables($plugintype = null) {
+
+        $templatevars = array();
+
+        if ($plugintype == 'enrol') {
+            $templatevars = array(
+                array('name' => 'allow_enrol', 'hint' => 'boolean', 'values' => array('True', 'False')),
```

Comment:
I've decided to use this format for the template variables mainly because of the web interface.

One possible use for the hint field is to have the value from modified by the web interface before passing it to the manager. For example, in this case where the hint is 'boolean', the value 'True' (string) will be replace by true (boolean). I can imagine other use cases too: we want the value passed to the manager to be a number, so we use the hint field for that.

Another use case for the hint would be in the manager: we can set the hint to 'strict', and if the user recipe doesn't have the expected values (from the 'values' array), then we throw an exception. Or we could set the hint to 'warning', and in this case we only log a warning.

The 'values' array represents the possible accepted values for the variable. We can use it in the manager, to validate the recipe, or in the web interface, to create a drop-down menu with these values.